src/main/java/org/springframework/data/elasticsearch/core/query/DeleteQuery.java

```diff
@@ -0,0 +1,733 @@
+
+    /**
+     * Maximum number of documents to process.
+     * Defaults to all documents.
+     */
+    @Nullable
+    private final Long maxDocs;
+
+    /**
+     * Specifies the node or shard the operation should be performed on.
+     */
+    @Nullable
+    private final String preference;
+
+    /**
+     * Use the request cache when it is {@code true}.
+     * By default, use the index-level setting.
+     */
+    @Nullable
+    private final Boolean requestCache;
+
+    /**
+     * Refreshes all shards involved in the deleting by query after the request completes when it is {@code true}.
+     * By default, this is set to {@code false}.
+     */
+    @Nullable
+    private final Boolean refresh;
+
+    /**
+     * Limited this request to a certain number of sub-requests per second.
+     * By default, this is set to {@code -1} (no throttle).
+     */
+    @Nullable
+    private final Float requestsPerSecond;
+
+    /**
+     * Custom value used to route operations to a specific shard.
+     */
+    @Nullable
+    private final String routing;
+
+    /**
+     * Period to retain the search context for scrolling.
+     */
+    @Nullable
+    private final Duration scroll;
+
+    /**
+     * Size of the scroll request that powers the operation.
+     * By default, this is set to {@code 1000}.
+     */
+    @Nullable
+    private final Long scrollSize;
+
+    /**
+     * The type of the search operation.
+     */
+    @Nullable
+    private final SearchType searchType;
+
+    /**
+     * Explicit timeout for each search request.
+     * By default, this is set to no timeout.
+     */
+    @Nullable
+    private final Duration searchTimeout;
+
+    /**
+     * The number of slices this task should be divided into.
+     * By default, this is set to {@code 1} meaning the task isn’t sliced into subtasks.
+     */
+    @Nullable
+    private final Integer slices;
+
+    /**
+     * Sort search results in a specific order.
+     */
+    @Nullable
+    private final Sort sort;
+
+    /**
+     * Specific {@code tag} of the request for logging and statistical purposes.
+     */
+    @Nullable
+    private final List<String> stats;
+
+    /**
+     * The Maximum number of documents that can be collected for each shard.
+     * If a query exceeds this limit, Elasticsearch will stop the query.
+     */
+    @Nullable
+    private final Long terminateAfter;
+
+    /**
+     * Period each deletion request waits for active shards.
+     * By default, this is set to {@code 1m} (one minute).
+     */
+    @Nullable
+    private final Duration timeout;
+
+    /**
+     * Returns the document version as part of a hit.
+     */
+    @Nullable
+    private final Boolean version;
+
+    // Body
+    /**
+     * Query that specifies the documents to delete.
+     */
+    private final Query query;
+
+    public static Builder builder(Query query) {
+        return new Builder(query);
+    }
+
+    public DeleteQuery(Builder builder) {
```

Comment:
```suggestion
    private DeleteQuery(Builder builder) {
```
no need to expose this, the `DeleteQuery` should be created using the builder.